src/pages/index.js

```diff
@@ -3,141 +3,201 @@ import "../pages/index.css";
+const handleCardFormSubmit = new FormValidator(config, addNewCardModal);
 
-editFormValidator.enableValidation();
-cardFormValidator.enableValidation();
+handleEditFormSubmit.enableValidation();
+handleProfileFormSubmit.enableValidation();
+handleCardFormSubmit.enableValidation();
 
 //FORM SUBMIT
 function handleFormSubmit(data) {
-  console.log(data)
-  userInfo.setUserInfo(data)
-  editProfile.close();
+  editProfile.setSubmitText(true, "Saving...");
+  api
+    .profileUpdate(data)
+    .then(() => {
+      userInfo.setUserInfo(data);
+      editProfile.setSubmitText(false);
+      editProfile.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+
 }
 
 // //IMAGE SUBMIT
-function handleImageFormSubmit() {
-  const name = newCardModalTitle.value;
-  const link = newCardModalUrl.value;
-  generateCard({ name, link });
-  openImagePopup.close();
+function handleImageFormSubmit(data) {
+  const name = data.name;
+  const link = data.url;
+  openImagePopup.setSubmitText(true, "Saving...");
+  api
+    .addCard(name, link)
+    .then((card) => {
+      generateCard(card);
+      openImagePopup.setSubmitText(false)
+      openImagePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+  
+}
+
+function profilePicEditSubmit(data) {
+  const url = data.url;
+  profilePicEdit.setSubmitText(true, "Loading...");
+  api
+    .updateProfileAvatar(url)
+    .then(() => {
+      profileImage.src = url;
+      profilePicEdit.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => profilePicEdit.setSubmitText(false));
+}
+
+//IMAGE LIKE
+function handleCardLike(card) {
+  const newLikeStatus = !card.isLiked;
+  if(newLikeStatus) {
+    api.cardLike(card.getId())
+    .then((res) => {
+      card.setLikeStatus(res.isLiked);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+  } else {
+    api.removeLike(card.getId())
+    .then((res) => {
+      card.setLikeStatus(res.isLiked);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+  }
 }
 
 //PROFILE EDIT POPUP
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
-//EDIT PROFILE
+//EVENT LISTENERS
 profileEditButton.addEventListener("click", () => {
-  profileModalName.value = profileName.textContent;
-  profileModalDescription.value = profileDescription.textContent.trim();
+  const currentUser = userInfo.getUserInfo();
+  profileModalName.value = currentUser.name;
+  profileModalDescription.value = currentUser.title;
   editProfile.open();
 });
 
-//EVENT LISTENERS
 addNewCardButton.addEventListener("click", () => {
-  cardFormValidator.toggleButtonState();
+  handleCardFormSubmit.toggleButtonState();
   openImagePopup.open();
 });
 
+profilePicButton.addEventListener("click", () => {
+  profilePicEdit.open();
```

Comment:
* The avatar submit button should be disabled after submitting. Try to change the avatar and open the modal again (the button is active then)


Please, call `toggleButtonState` here


```js

avatarFormValidator.toggleButtonState()
```